### PR TITLE
chore: show logger messages in tests/ when -verbose with gnodev test

### PR DESCRIPTION
**Pull request by grepsuzette**
While working on https://github.com/gnolang/gno/issues/615

`f.logger(<...>)` calls in tests/file.go are not shown when `gnodev test -verbose (...)`. 

If the cmd user is specifying -verbose (and since we don't have -debug ATM) it seems those traces should be shown.

For instance this is called from .tests/file.go:
```
    if f.logger != nil {
            store.Print()
            f.logger("========================================")
            f.logger("PREPROCESS ALL FILES")
            f.logger("========================================")
    }
```

# How has this been tested?

With `gnodev test -verbose (...)`.

### CODE REVIEW

**Comment by zivkovicmilos** [comment]
@grepsuzette

Thank you for opening this PR 🙏 

I'm not sure if this is the right approach - to bundle log lines with the `verbose` flag, as the `verbose` flag has the same semantics as `go test -v` (just shows the tests being run).

Do you think adding a logger flag would be more appropriate?

cc @moul

**Comment by moul** [comment]
I think you should go ahead with this approach, then come back to a previous discussion on having better filtering support globally.

File: cmd/gnodev/test.go [hunk 1]
```diff
@@ -291,7 +291,13 @@ func gnoTestPkg(
 			}
 
 			testFilePath := filepath.Join(pkgPath, testFileName)
-			err := tests.RunFileTest(rootDir, testFilePath, tests.WithSyncWanted(cfg.updateGoldenTests))
+			var fLogger func(args ...interface{}) = nil // note: tests.loggerFunc is not exported...
```
Comment (zivkovicmilos) [inline cmd/gnodev/test.go:294]: Why not export `tests.loggerFunc`? It's an option anyways (`WithLoggerFunc`)

**Review by zivkovicmilos — APPROVED** [review]
Looks good 💯 

Let's go with @moul's feedback, and circle back to discussions later on if needed

File: cmd/gnodev/test.go (hunk 1 above)
Comment (grepsuzette) [reply cmd/gnodev/test.go:294]: By export do you mean changing `loggerFunc` -> `LoggerFunc` in the other package right?
As I'm still a new contributor I try to operate chirurgically but I admit what you say is more logical :)
Thanks

Comment (zivkovicmilos) [reply cmd/gnodev/test.go]: Because you exported it now, you can use it as `var fLogger LoggerFunc` in `cmd/gnodev/test.go`

**Review by zivkovicmilos — COMMENTED** [review]
@grepsuzette 

You have a failing test in the `test4` suite:

`FAIL: TestTest/test_--verbose_--precompile_..~..~tests~integ~failing2`

It's probably due to you changing the expected output in this PR. Please resolve it before we can merge this 🙏

**Comment by grepsuzette** [comment]
> @grepsuzette do you need help with fixing tests?

The `Go Test (1.20.x, test.examples)/4_test.txt` report is too long to be shown on github.
I've downloaded their zip. It contains no fail or FAIL. 
My impression for now is that maybe the test ran for too long (19 min).

I am try to run it locally (with `go run ./cmd/gnodev test --verbose ./examples`) while measuring the time it takes.

**Comment by grepsuzette** [comment]
tl;dr: 
Like @zivkovicmilos said, if `-verbose` influences unit tests, it's not the best way for sure. Maybe we are after a `-debug` option or something. 

Below you will see 3 tests for test suite 4. 

1. local test, -verbose
2. local test, not verbose
3. local test with a version 1 commit *before* this commit (`1deeb1`).

And I see no problem other than it takes slightly more time.
But I don't think this problem is worth anyone's time. Therefore maybe we want to close it.

--

The first local test ended with:
```
        Realm:
          gno.land/r/demo/users
        Exception:
          ("invalid user g1w3jhxapjta047h6lta047h6lta047h6laqcyu4" string)
          invalid user g1w3jhxapjta047h6lta047h6lta047h6laqcyu4
OUTPUT:

--- PASS: file/z_8_filetest.gno (2.00s)
=== RUN   file/z_9_filetest.gno
========================================
RUN FILES & INIT
========================================
========================================
RUN MAIN
========================================
========================================
RUN MAIN END
========================================
--- PASS: file/z_9_filetest.gno (1.97s)
ok      ./examples/gno.land/r/demo/users        32.60s
=== RUN   TestRender
--- PASS: TestRender (0.01s)
ok      ./examples/gno.land/r/demo/x/outfmt     2.40s
ok      ./examples/gno.land/r/demo/x/upgrade/upgrade-a  1.96s
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-a/v1       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-a/v2       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-b/v1       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-b/v2       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-c/root     [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-c/v1       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-c/v2       [no test files]
=== RUN   TestPackage
--- PASS: TestPackage (0.02s)
ok      ./examples/gno.land/r/gnoland/blog      2.52s
=== RUN   TestPackage
--- PASS: TestPackage (0.01s)
ok      ./examples/gno.land/r/gnoland/faucet    2.03s
?       ./examples/gno.land/r/system/names      [no test files]
?       ./examples/gno.land/r/system/rewards    [no test files]
?       ./examples/gno.land/r/system/validators         [no test files]

real    5m38.517s
user    14m34.567s
sys     0m30.781s
```

`echo $?` gives `0`.
Ran for about 5 minutes.
So I see no problem here. 

---

Then tried to run it without -verbose, same thing:
```
(...)
ok      ./examples/gno.land/r/demo/x/outfmt     2.34s
ok      ./examples/gno.land/r/demo/x/upgrade/upgrade-a  1.92s
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-a/v1       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-a/v2       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-b/v1       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-b/v2       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-c/root     [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-c/v1       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-c/v2       [no test files]
ok      ./examples/gno.land/r/gnoland/blog      2.46s
ok      ./examples/gno.land/r/gnoland/faucet    2.02s
?       ./examples/gno.land/r/system/names      [no test files]
?       ./examples/gno.land/r/system/rewards    [no test files]
?       ./examples/gno.land/r/system/validators         [no test files]

real    4m8.704s
user    13m42.285s
sys     0m21.293s
```
Time about the same.

---

Next I will run the same test *without* the PR, resetting to the immediate commit which seems `1deeb1`:

```
        Realm:
          gno.land/r/demo/users
        Exception:
          ("invalid user g1w3jhxapjta047h6lta047h6lta047h6laqcyu4" string)
          invalid user g1w3jhxapjta047h6lta047h6lta047h6laqcyu4
OUTPUT:
 
--- PASS: file/z_8_filetest.gno (2.10s)
=== RUN   file/z_9_filetest.gno
--- PASS: file/z_9_filetest.gno (2.02s)
ok      ./examples/gno.land/r/demo/users        28.52s
=== RUN   TestRender
--- PASS: TestRender (0.01s)
ok      ./examples/gno.land/r/demo/x/outfmt     2.46s
ok      ./examples/gno.land/r/demo/x/upgrade/upgrade-a  1.98s
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-a/v1       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-a/v2       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-b/v1       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-b/v2       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-c/root     [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-c/v1       [no test files]
?       ./examples/gno.land/r/demo/x/upgrade/upgrade-c/v2       [no test files]
=== RUN   TestPackage
--- PASS: TestPackage (0.01s)
ok      ./examples/gno.land/r/gnoland/blog      2.48s
=== RUN   TestPackage
--- PASS: TestPackage (0.01s)
ok      ./examples/gno.land/r/gnoland/faucet    2.21s
?       ./examples/gno.land/r/system/names      [no test files]
?       ./examples/gno.land/r/system/rewards    [no test files]
?       ./examples/gno.land/r/system/validators         [no test files]

real    4m1.422s
user    13m18.233s
sys     0m21.298s
```

I see no difference except the test runs in 4min1sec instead of 5min38.

---
I don't know if it's a timer problem or otherwise. 

Like I said in the tl;dr, I was not aware of the testing system before I proposed this PR, so I am not comfortable with this approach anymore. 

As for a `-debug` switch for `gnodev` for this kind of developer-only traces not affecting the traces, it's up to you guys. My opinion is probably it doesn't cost anything and it's something that we want, but are there better alternatives... it's a question.

**Comment by ajnavarro** [comment]
@grepsuzette could you rebase and get the latest changes from master? thanks!

**Comment by thehowl** [comment]
Closing as it's been a draft for 1yr+.
